mitosheet/mitosheet/parser.py

```diff
@@ -256,69 +420,110 @@ def find_column_headers(match: Any) -> Any:
             # If this column header was already covered by another column header
             # that has been found, then this column header is just a substring
             # of another column header, so we avoid matching it
-            if match_covered_by_matches([match for _, match in column_header_match_tuples], match):
+            if match_covered_by_matches([match['match_range'] for match in parser_matches], match_range):
                 return found_column_header
 
-            # Do not replace if it is part of a function, which means it has
-            # another ascii character before or after it. Or if it is part of
-            # a number. Or if it has a ( after it, then it's a function call
-            
-            if (start - 1 >= 0 and formula[start - 1].isalnum()) or \
-                (end < len(formula) and (formula[end].isalnum() or formula[end] == '(')):
-                return found_column_header
+            # First, we check if it's an unqualified column header with no index. This is just 
+            # if the characters after the header are 
+            if is_column_header_without_index(formula, index, start, end):
+                parser_matches.append({
+                    'type': COLUMN_HEADER_MATCH_TYPE,
```

Comment:
Maybe we should call this type `COLUMN_HEADER_WITHOUT_INDEX_MATCH_TYPE` . The name `COLUMN_HEADER_MATCH_TYPE` doesn't really tell the full story.